[BUG] SOF tools build fails

**Describe the bug**

I'm using Ubuntu 19.10 and I get this with GCC version 9.2.1

```
singalsu@singalsu-MOBL:~/work/current/sof/sof/tools/build_tools$ make
[  0%] Building C object probes/CMakeFiles/sof-probes.dir/probes_main.c.o
/home/singalsu/work/current/sof/sof/tools/probes/probes_main.c: In function ‘parse_data’:
/home/singalsu/work/current/sof/sof/tools/probes/probes_main.c:222:5: error: converting a packed ‘struct probe_data_packet’ pointer (alignment 1) to a ‘uint32_t’ {aka ‘unsigned int’} pointer (alignment 4) may result in an unaligned pointer value [-Werror=address-of-packed-member]
  222 |     w_ptr = (uint32_t *)packet;
      |     ^~~~~
In file included from /home/singalsu/work/current/sof/sof/tools/probes/probes_main.c:16:
/home/singalsu/work/current/sof/sof/tools/probes/../../src/include/ipc/probe.h:88:8: note: defined here
   88 | struct probe_data_packet {
      |        ^~~~~~~~~~~~~~~~~
/home/singalsu/work/current/sof/sof/tools/probes/probes_main.c:254:26: error: taking address of packed member of ‘struct probe_data_packet’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
  254 |      w_ptr = (uint32_t *)&packet->data;
      |                          ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [probes/CMakeFiles/sof-probes.dir/build.make:63: probes/CMakeFiles/sof-probes.dir/probes_main.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: probes/CMakeFiles/sof-probes.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

**To Reproduce**
Run scripts/build-tools.sh

**Reproduction Rate**
How often does the issue happen ? i.e. 1/10 (once in ten attempts), 1/1000 or all the time.
Does the reproduction rate vary with any other configuration or user action, if so please describe and show the new reproduction rate.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel: {SHA}
    * SOF: {SHA}
2) Name of the topology file
    * Topology: {FILE}
3) Name of the platform(s) on which the bug is observed.
    * Platform: {PLATFORM}

**Screenshots or console output**
If applicable, add a screenshot (drag-and-drop an image), or console logs
(cut-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue.

Please also include the relevant sections from the firmware log and kernel log in the report (and attach the full logs for complete reference). Kernel log is taken from *dmesg* and firmware log from *sof-logger*. See https://thesofproject.github.io/latest/developer_guides/debugability/logger/index.html

--- a/tools/probes/probes_main.c
+++ b/tools/probes/probes_main.c
@@ -199,6 +199,12 @@ void parse_data(char *file_in)
 	}
 
 	packet = malloc(PACKET_MAX_SIZE);
+	if (!packet) {
+		fprintf(stderr, "error: allocation failed, err %d\n",
+			errno);
+		fclose(fd_in);
+		exit(0);
+	}
 	memset(&data, 0, sizeof(uint32_t) * DATA_READ_LIMIT);
 	memset(&files, 0, sizeof(struct wave_files) * FILES_LIMIT);
 

--- a/tools/probes/probes_main.c
+++ b/tools/probes/probes_main.c
@@ -219,7 +219,8 @@ void parse_data(char *file_in)
 				/* request to copy full data packet */
 				total_data_to_copy = sizeof(struct probe_data_packet) /
 					sizeof(uint32_t);
-				w_ptr = (uint32_t *)packet;
+				/* probe_data_packet forced to align 4 */
+				w_ptr = __builtin_assume_aligned((uint32_t *)packet, 4);
 				state = SYNC;
 			}
 			/* data copying section */
@@ -251,7 +252,8 @@ void parse_data(char *file_in)
 						packet = realloc(packet,
 								 sizeof(struct probe_data_packet) +
 								 packet->data_size_bytes);
-					w_ptr = (uint32_t *)&packet->data;
+					/* probe_data_packet forced to align 4 */
+					w_ptr = __builtin_assume_aligned((uint32_t *)&packet->data, 4);
 					state = CHECK;
 					break;
 				case CHECK:

